src/ActionList/List.tsx

```diff
@@ -162,7 +164,7 @@ export const List = React.forwardRef<HTMLDivElement, ListProps>((props, forwarde
   const renderItem = (itemProps: ItemInput, item: ItemInput, itemIndex: number) => {
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     const ItemComponent = ('renderItem' in itemProps && itemProps.renderItem) || props.renderItem || Item
-    const key = itemProps.key ?? itemProps.id?.toString() ?? itemIndex.toString()
+    const key = itemProps.id?.toString() ?? itemIndex.toString()
```

Comment:
Non-blocking: Out of curiosity, why was this change to `key` necessary?

Reply:
Good catch — this was my mistake. TS caught that `key` wasn't on `ItemInput` so I removed the line intending to circle back to it. Will fix!